docs/docs/design/candidate-backing-elastic-scaling.md

```diff
@@ -0,0 +1,116 @@
+# Candidate backing subsystem design update
+> Update Candidate backing subsystem to support Async backing + Elastic scaling
+
+## overview of changes
+### Remove use of ProspectiveParachainsMode
+We currently have both legacy backing code and async backing code. Now that the async backing parameters runtime API is live on all networks, the backing subsystem code can be simplified. We can remove the use of `ProspectiveParachainsMode` and keep only the parts under `ProspectiveParachainsMode.Enabled`
+
+### remove perLeaf field of CandidateBacking struct
+- Remove the perLeaf field of the `Candidatebacking` struct  as it keeps track of which Active leaf has ProspectiveParachainMode Enabled.
+- if we need active leaf data, we can get it from backing implicit view which we store in `CandidateBacking`. (Yet to implement)
+
+### change GroupID from paraID to CoreIndex
+Before elastic scaling, only a single group of validators was assigned to a parachain, so we assigned paraID to GroupID.
+![single-group-assignment](../assets/img/single-group-assignment.png)
+
+However, in elastic scaling, multiple groups of validators(core) will be assigned to a parachain. So now we have to change the type of GroupID from paraID to the `CoreIndex`, which is unique for assigned groups.
+![multiple-group-assignment](../assets/img/multiple-group-assignment.png)
+
+
+- Change the type of groupID from `paraID` to `CoreIndex` and check all the methods to see if any changes are needed.
+    - Structs containing groupID: `Summary`, `tableContext`, `attestedCandidate`, `candidateData`.
+- change `assignment *parachaintypes.ParaID` field to `assignedCore *parachaintypes.CoreIndex` in perRelayParentState which represents the core index assigned to the local validator at this relay parent.
+
+### Use fragment-chain
+Our async backing code uses `Fragment-tree`, which was introduced in Async backing and has now become `Fragment-chain`. So we need to update the code to use `Fragment-chain`.
+- TODO: What/Where/How to update the code?
```

Comment:
Are you going to address this before merging?

Reply:
I don't think. That would be an implementation detail.
So, I will remove this to-do.